front/pages/w/[wId]/assistant/labs/transcripts/index.tsx

```diff
@@ -84,32 +90,39 @@ export default function LabsTranscriptsIndex({
     transcriptsConfiguration,
     isTranscriptsConfigurationLoading,
     mutateTranscriptsConfiguration,
-  } = useLabsTranscriptsConfiguration({
-    workspaceId: owner.sId,
-    provider,
-  });
+  } = useLabsTranscriptsConfiguration({ workspaceId: owner.sId });
 
   const [transcriptsConfigurationState, setTranscriptsConfigurationState] =
     useState<{
+      provider: string;
       isGDriveConnected: boolean;
+      isGongConnected: boolean;
       assistantSelected: LightAgentConfigurationType | null;
       isActive: boolean;
     }>({
+      provider: transcriptsConfiguration?.provider || "",
       isGDriveConnected: false,
-      assistantSelected: null as LightAgentConfigurationType | null,
+      isGongConnected: false,
+      assistantSelected: null,
       isActive: false,
     });
 
   useEffect(() => {
-    setTranscriptsConfigurationState({
-      isGDriveConnected:
-        (transcriptsConfiguration && transcriptsConfiguration.id > 0) || false,
-      assistantSelected:
-        agentConfigurations.find(
-          (a) => a.sId === transcriptsConfiguration?.agentConfigurationId
-        ) || null,
-      isActive: transcriptsConfiguration?.isActive || false,
-    });
+    if (transcriptsConfiguration) {
+      setTranscriptsConfigurationState({
```

Comment:
When updating a state object using the previous state, you must use the following form to  avoid race condition bugs:
```
setTranscriptsConfigurationState((prev) => {
    return {...prev, modification:modification}
})
```

Reply:
Interesting, didn't know! Changed.